tests/io/test_core.py

```diff
@@ -61,49 +219,172 @@ def test_get_filepath_str(self):
-                    "path": "mycontainer@mystorageaccount.dfs.core.windows.net/mypath",
-                },
-            ),
-            (
-                "hdfs://namenode:8020/file.txt",
-                {"protocol": "hdfs", "path": "/file.txt"},
-            ),
-            ("file:///tmp/file.txt", {"protocol": "file", "path": "/tmp/file.txt"}),
-            ("/tmp/file.txt", {"protocol": "file", "path": "/tmp/file.txt"}),
-            (
-                "C:\\Projects\\file.txt",
-                {"protocol": "file", "path": "C:\\Projects\\file.txt"},
-            ),
-            (
-                "file:///C:\\Projects\\file.txt",
-                {"protocol": "file", "path": "C:\\Projects\\file.txt"},
-            ),
-            (
-                "https://example.com/file.txt",
-                {"protocol": "https", "path": "https://example.com/file.txt"},
-            ),
-            (
-                "http://example.com/file.txt",
-                {"protocol": "http", "path": "http://example.com/file.txt"},
+                ("abfss", "mycontainer@mystorageaccount.dfs.core.windows.net/mypath"),
             ),
+            ("hdfs://namenode:8020/file.txt", ("hdfs", "/file.txt")),
+            ("file:///tmp/file.txt", ("file", "/tmp/file.txt")),
+            ("/tmp/file.txt", ("file", "/tmp/file.txt")),
+            ("C:\\Projects\\file.txt", ("file", "C:\\Projects\\file.txt")),
+            ("file:///C:\\Projects\\file.txt", ("file", "C:\\Projects\\file.txt")),
+            ("https://example.com/file.txt", ("https", "example.com/file.txt")),
+            ("http://example.com/file.txt", ("http", "example.com/file.txt")),
         ],
     )
-    def test_parse_filepath(self, filepath, expected_result):
-        assert _parse_filepath(filepath) == expected_result
+    def test_get_protocol_and_path(self, filepath, expected_result):
+        assert get_protocol_and_path(filepath) == expected_result
+
+    @pytest.mark.parametrize(
+        "filepath",
+        [
+            "http://example.com/file.txt",
+            "https://example.com/file.txt",
+        ],
+    )
+    def test_get_protocol_and_path_http_with_version(self, filepath):
+        version = Version(load=None, save=None)
+        expected_error_message = "Versioning is not supported for HTTP protocols. Please remove the `versioned` flag from the dataset configuration."
+        with pytest.raises(DatasetError, match=expected_error_message):
+            get_protocol_and_path(filepath, version)
+
+    @pytest.mark.parametrize(
+        "input", [{"key1": "invalid value"}, {"key2": "invalid;value"}]
+    )
+    def test_validate_forbidden_chars(self, input):
+        key = list(input.keys())[0]
+        expected_error_message = (
+            f"Neither white-space nor semicolon are allowed in '{key}'."
+        )
+        with pytest.raises(DatasetError, match=expected_error_message):
+            validate_on_forbidden_chars(**input)
+
+
+class TestAbstractDataSet:
+    pass
```

Comment:
What is this for?

Reply:
Probably left over from something Ahdra was trying. Reverting both these things.